sympy/core/exprtools.py

```diff
@@ -355,6 +354,12 @@ def __init__(self, factors=None):  # Factors
             for _ in range(i):
                 c.remove(I)
             factors = dict(Mul._from_args(c).as_powers_dict())
+            # Handle all rational Coefficients
+            for f in list(factors.keys()):
+                if type(f) is Rational or type(f) is Half:
```

Comment:
This could be `if isinstance(f, Rational)`

Reply:
And then `Half won't need importing above.

Reply:
Did both of those things.